streaming/sender.c

```diff
@@ -717,7 +717,7 @@ void *rrdpush_sender_thread(void *ptr) {
     s->timeout = (int)appconfig_get_number(&stream_config, CONFIG_SECTION_STREAM, "timeout seconds", 60);
     s->default_port = (int)appconfig_get_number(&stream_config, CONFIG_SECTION_STREAM, "default port", 19999);
     s->buffer->max_size =
-        (size_t)appconfig_get_number(&stream_config, CONFIG_SECTION_STREAM, "buffer size bytes", 1024 * 1024);
+        (size_t)appconfig_get_number(&stream_config, CONFIG_SECTION_STREAM, "buffer size bytes", 1024 * 1024 * 10);
```

Comment:
For clarity and better readability I would expect to see this change also in `line 686`
`s->buffer = cbuffer_new(1024, 1024*1024*10);`

Reply:
I was under impression that the initial size is supposed to be 1MB because we don't change it regardless of the "buffer size bytes" value.

Reply:
This is what you want to achieve?
1. Initial buffer size is 1024 bytes.
2. Initial max size is 1MB.
3. Max size after configuration is 10MB.

Reply:
This is the current logic (or my understanding of it):
 - the initial buffer size is always 1MB

https://github.com/netdata/netdata/blob/549e4621a9cd1cbc212ec8620ebd4afcc031fb81/streaming/sender.c#L683-L686

- we increase it up to max (which is `buffer size bytes`) in 

https://github.com/netdata/netdata/blob/549e4621a9cd1cbc212ec8620ebd4afcc031fb81/streaming/sender.c#L74-L75

---

> This is what you want to achieve?

I just bumped `buffer size bytes`. I am not against changing the initial buffer size, just saying that I haven't done because I thought that is the idea - to start with 1MB and grow (if neede) to max.

Reply:
Code will do what you expected so I am going to approve it. 👍🏼 

Note:
`s->buffer = cbuffer_new(1024, 1024*1024);` means that it starts with a 1024bytes (not MB) and can increase up to max size of 1MB every time the cbuffer_add_unsafe() is called (call to this function doubles the size of the buffer until the max size).
So the current code what will do is,
1. Initial buffer size is 1024 bytes.
2. Initial max size is 1MB (at host creation).
3. Update max size after configuration to 10MB (at sender thread call).

So between 2. and 3. the buffer will not have the configuration default value but the 1MB value this is the reason I made this comment. 😄